kstreams-live-update/aggregator/src/main/java/io/debezium/examples/kstreams/liveupdate/aggregator/StreamsPipeline.java

```diff
@@ -17,57 +17,65 @@
+        Serde<Order> temperatureMeasurementsSerde = new ChangeEventAwareJsonSerde<>(Order.class);
         temperatureMeasurementsSerde.configure(Collections.emptyMap(), false);
 
-        Serde<Station> stationSerde = new ChangeEventAwareJsonSerde<>(Station.class);
+        Serde<Category> stationSerde = new ChangeEventAwareJsonSerde<>(Category.class);
         stationSerde.configure(Collections.emptyMap(), false);
 
-        Serde<CountAndSum> countAndSumSerde = new ChangeEventAwareJsonSerde<>(CountAndSum.class);
+        Serde<ValueAggregator> countAndSumSerde = new ChangeEventAwareJsonSerde<>(ValueAggregator.class);
         stationSerde.configure(Collections.emptyMap(), false);
 
-        KTable<Long, Station> stations = builder.table("dbserver1.inventory.stations", Consumed.with(longKeySerde, stationSerde));
+        KTable<Long, Category> stations = builder.table("dbserver1.inventory.categories", Consumed.with(longKeySerde, stationSerde));
 
         return builder.stream(
-                "dbserver1.inventory.temperature_measurements",
+                "dbserver1.inventory.orders",
                 Consumed.with(longKeySerde, temperatureMeasurementsSerde)
                 )
-                .selectKey((k, v) -> v.stationId)
+                .selectKey((k, v) -> v.categoryId)
                 .join(
                         stations,
                         (value1, value2) -> {
-                            value1.stationName = value2.name;
+                            value1.categoryName = value2.name;
                             return value1;
                         },
                         Joined.with(Serdes.Long(), temperatureMeasurementsSerde, null)
                 )
-                .selectKey((k, v) -> v.stationName)
+                .selectKey((k, v) -> v.categoryName)
                 .groupByKey(Serialized.with(Serdes.String(), temperatureMeasurementsSerde))
+                .windowedBy(TimeWindows.of(1000))
                 .aggregate(
-                        () -> new CountAndSum(), /* initializer */
+                        () -> new ValueAggregator(), /* initializer */
                         (aggKey, newValue, aggValue) -> {
-                            aggValue.count++;
-                            aggValue.sum += newValue.value;
-                            return aggValue;
+                            return aggValue.increment(newValue.salesPrice);
                         },
                         Materialized.with(Serdes.String(), countAndSumSerde)
                 )
-                .mapValues(v -> {
-                    BigDecimal avg = BigDecimal.valueOf(v.sum / (double) v.count);
-                    avg = avg.setScale(1, RoundingMode.HALF_UP);
-                    return avg.doubleValue();
-                })
+                .mapValues(v -> BigDecimal.valueOf(v.sum)
+                        .divide(BigDecimal.valueOf(v.count * 100), 2, RoundingMode.HALF_UP))
```

Comment:
Why is there the multiplication by 100?

Reply:
It's because the values are stored as "cents" in the DB, e.g. 1234 for $12.34.